### DISCUSSION TITLE
[mini.pick] Is it possible to get the file name at the start?

### DISCUSSION THREAD

**Discussion by lsig** [nvim-mini/mini.nvim#1082] [Q&A] [answered]
Hi, I was wondering if it is possible to get the file names first and have the directories come after for Pick *files*? 

fzf-lua:
![image](https://github.com/user-attachments/assets/6bc2d05d-fb3d-44df-9d5e-b6d08c941533)

MiniPick:
![image](https://github.com/user-attachments/assets/5ad1b09f-8f1f-4e24-9ead-f55faee2b503)

**Comment by echasnovski** [comment, marked as answer]
This should be possible by overriding [`source.show()`](https://github.com/echasnovski/mini.nvim/blob/657ef75b52a548090fb7906735517b7ddbf39e88/doc/mini-pick.txt#L353) for picker you want this behavior (presumably, `builtin.files`).
However, I think it will be a tough challenge to do *while* preserving all features of [`default_show()`](https://github.com/echasnovski/mini.nvim/blob/657ef75b52a548090fb7906735517b7ddbf39e88/doc/mini-pick.txt#L938). Primarily, adding icons and current/match/marked highlighting.

As I think showing matched characters is really important and having file name first makes it at least very complex, I don't think this will be made built-in or showed here in full capability. Here is an example of how the backbone can look like:

```lua
local show_filename_first = function(buf_id, item_arr, query)
  local lines = {}
  for i, item in ipairs(item_arr) do
    local directory, basename = vim.fs.dirname(item), vim.fs.basename(item)
    lines[i] = string.format('%s (%s)', basename, directory)
  end
  vim.api.nvim_buf_set_lines(buf_id, 0, -1, false, lines)
end
MiniPick.registry.files_2 = function() MiniPick.builtin.files({}, { source = { show = show_filename_first } }) end
```

You can then use it with `:Pick files_2`.

**Reply by lsig** [reply]
Alright, thanks for your help :)
